mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapper.java

```diff
@@ -130,38 +121,26 @@ public CassandraMessageIdMapper(MailboxMapper mailboxMapper, CassandraMailboxDAO
                         .completableFuture());
     }
 
-    private CompletableFuture<Boolean> mailboxExists(MessageWithoutAttachment messageWithoutAttachment) {
-        CassandraId cassandraId = (CassandraId) messageWithoutAttachment.getMailboxId();
+    private CompletableFuture<Stream<SimpleMailboxMessage>> filterMessagesWIthExistingMailbox(Stream<SimpleMailboxMessage> stream) {
+        return FluentFutureStream.of(stream.map(this::mailboxExists))
+            .flatMap(m -> m)
+            .completableFuture();
+    }
+
+    private CompletableFuture<Stream<SimpleMailboxMessage>> mailboxExists(SimpleMailboxMessage message) {
```

Comment:
Why a Stream if you only return one message, this is really confusing. You should return a Optional and than convert this Optional into a Stream using OptionalConverter.toStream (line 125).

Also I don't like the name of your method because you don't return true or false to know if the mailboxExists to be you return the actual mailbox. Therefore I will name this method "getMailboxIfExists".
But after doing all of that you method will have no extra value compare to message.getMailboxId() except logging. Maybe it's not worst it extracting it to a private method.